lib/db/index.js

```diff
@@ -37,7 +38,11 @@ const db = require('knex')({
     }
 
     if (timeseries) {
-      postProcessed = createCompleteTimeseries({ data: postProcessed || data, metric, startDate, endDate, interval })
+      postProcessed = createCompleteTimeseries({ data: postProcessed || data, metric, ...timeseries })
+    }
+
+    if (transposeAndAggregate) {
+      postProcessed = transposeAndAggregateByDimensions({ data: postProcessed || data, metric, timeDimension, dimensions })
     }
```

Comment:
Should this be an if/else-if

Reply:
No, because it can be a time-series as well as a transpose and aggregate query.